source/gloperate/include/gloperate/primitives/PolygonalGeometry.h

```diff
@@ -126,11 +126,66 @@ class GLOPERATE_API PolygonalGeometry
     */
     void setNormals(std::vector<glm::vec3> && normals);
 
+	/**
+	*  @brief
+	*    Check if mesh contains texture coordinates
+	*
+	*  @return
+	*    'true' if the mesh contains texture coordinates, otherwise 'false'
+	*/
+	bool hasTextureCoordinates() const;
+
+	/**
+	*  @brief
+	*    Get texture coordinate array
+	*
+	*  @return
+	*    Texture coordinate array
+	*/
+	const std::vector<glm::vec3> & textureCoordinates() const;
+
+	/**
+	*  @brief
+	*    Set texture coordinates
+	*
+	*  @param[in] textureCoordinates
+	*    Texture coordinate array
+	*/
+	void setTextureCoordinates(const std::vector<glm::vec3> & textureCoordinates);
+
+	/**
+	*  @brief
+	*    Set texture coordinates
+	*
+	*  @param[in] textureCoordinates
+	*    Texture coordinate array
+	*/
+	void setTextureCoordinates(std::vector<glm::vec3> && textureCoordinates);
+
+	/**
+	*  @brief
+	*    Get material index
+	*
+	*  @return
+	*    Material index
+	*/
+	unsigned int materialIndex() const;
+
+	/**
+	*  @brief
+	*    Set material index
+	*
+	*  @param[in] materialIndex
+	*    Material index
+	*/
+	void setMaterialIndex(const unsigned int materialIndex);
```

Comment:
For the declaration (mainly in the header file), the const for always copied types is both not required and sometimes punished with a warning. You should remove it from the declaration. But then again, you are encouraged to use the additional const in the definition of the function / method (this is actually possible, allowed, and serves the exact purpose I described here.